testdata/main.go

```diff
@@ -6,32 +6,122 @@ import (
 	"time"
 	"context"
 	"strings"
+	"os"
+	"encoding/json"
 
 	a "github.com/ibmdb/go_ibm_db"
 )
 
 var ctx = context.Background()
 
-var host = "<HOST>"
-var port = "<PORT>"
-var database = "<DATABASE>"
-var uid = "<UID>"
-var pwd = "<PWD>"
+var host string
+var port string
+var database string
+var uid string
+var pwd string
+var connStr string
 
-var connStr = "PROTOCOL=tcpip;HOSTNAME=" + host + ";PORT=" + port + ";DATABASE=" + database + ";UID=" + uid + ";PWD=" + pwd
+//Read Config variable from json file
+type Config  struct {
+        Host string `json:"HOSTNAME"`
+        Port string `json:"PORT"`
+	Database string `json:"DATABASE"`
+	Uid string `json:"UID"`
+	Pwd string `json:"PWD"`
+}
+
+func LoadConfiguration(filename string) (Config, error) {
+        var config Config
+        configFile, err := os.Open(filename)
+        defer configFile.Close()
+        if err != nil {
+                return config, err
+        }
+        jsonParser := json.NewDecoder(configFile)
+        err = jsonParser.Decode(&config)
+        return config, err
+}
+
+//Get connection information from config.json
+func GetConnectionInfoFromConfigFile() {
+       config, _:= LoadConfiguration("config.json")
+       host = config.Host
+       port = config.Port
+       database = config.Database
+       uid = config.Uid
+       pwd =  config.Pwd
+}
+
+
+//Get connection information from environment variables 
+func UpdateConnectionVariables() {
+	var databaseFound bool
+	var hostFound bool
+	var portFound bool
+	var uidFound bool
+	var pwdFound bool
+
+        config, _:= LoadConfiguration("config.json")
+
+	database, databaseFound = os.LookupEnv("DB2_DATABASE")
+        if !databaseFound{
+		database = config.Database
+		if len(database) == 0 {
+		    fmt.Println("Warning: Environment variable DB2_DATABASE is not set.")
+	        }
+	}
+
+	host, hostFound = os.LookupEnv("DB2_HOSTNAME")
+        if !hostFound{
+		host = config.Host
+		if len(host)==0 {
+		    fmt.Println("Warning: Environment variable DB2_HOSTNAME is not set.")
+	        }
+	}
+
+	port, portFound = os.LookupEnv("DB2_PORT")
+        if !portFound{
+		port = config.Port
+		if len(port)==0 {
+		    fmt.Println("Warning: Environment variable DB2_PORT not set.")
+	        }
+	}
 
+        uid, uidFound = os.LookupEnv("DB2_USER")
+        if !uidFound{
+		uid = config.Uid
+		if len(uid)==0 {
+		    fmt.Println("Warning: Environment variable DB2_USER is not set.")
+	        }
+	}
+
+        pwd, pwdFound = os.LookupEnv("DB2_PASSWD")
+        if !pwdFound{
+		pwd = config.Pwd
+
+		if len(pwd)==0 {
```

Comment:
@vmathur12 Remove this if check. We need to print warning for password if pwd is found in config file but env var DB2_USER is not set. Current code will not print warning if pwd is found in config file.

Reply:
@bimalkjha  Removed if len(pwd)==0 check. Please review and merge it. Thanks